core-bundle/src/Command/UserListCommand.php

```diff
@@ -0,0 +1,155 @@
+use Symfony\Component\Console\Style\SymfonyStyle;
+
+/**
+ * Lists Contao back end users.
+ *
+ * @internal
+ */
+class UserListCommand extends Command
+{
+    protected static $defaultName = 'contao:user:list';
+
+    /**
+     * @var ContaoFramework
+     */
+    private $framework;
+
+    public function __construct(ContaoFramework $framework)
+    {
+        $this->framework = $framework;
+
+        parent::__construct();
+    }
+
+    protected function configure(): void
+    {
+        $this
+            ->addOption('column', null, InputOption::VALUE_REQUIRED | InputOption::VALUE_IS_ARRAY, 'The columns display in the table')
+            ->addOption('admins', null, InputOption::VALUE_NONE, 'Return only admins')
+            ->addOption('format', null, InputOption::VALUE_REQUIRED, 'The output format', 'text')
+            ->setDescription('Lists Contao back end users.')
+        ;
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output): int
+    {
+        $io = new SymfonyStyle($input, $output);
+
+        if ($input->getOption('admins')) {
+            $io->note('Only showing admin accounts');
+        }
+
+        $users = $this->getUsers($input->getOption('admins'));
+
+        if (null === $users) {
+            $io->error('No accounts found.');
+
+            return 1;
+        }
+
+        $columns = $input->getOption('column');
+
+        switch ($input->getOption('format')) {
+            case 'text':
+                $rows = $this->formatTableRows($users, $columns);
+
+                $io->table($columns, $rows);
+                break;
+
+            case 'json':
+                $data = $this->formatJson($users, $columns);
+
+                $io->write(json_encode($data));
+                break;
+
+            default:
+                throw new \LogicException('What kind of crazy format is that!?');
+        }
+
+        return 0;
+    }
+
+    private function getUsers(bool $onlyAdmins = false): ?Collection
+    {
+        $this->framework->initialize();
+
+        /** @var UserModel $userModel */
+        $userModel = $this->framework->getAdapter(UserModel::class);
+
+        if ($onlyAdmins) {
+            return $userModel->findBy('admin', '1');
+        }
+
+        return $userModel->findAll();
+    }
+
+    private function formatTableRows(Collection $users, array &$columns): array
+    {
+        if ([] === $columns) {
+            $columns = ['username', 'name', 'admin', 'dateAdded', 'lastLogin'];
+        }
+
+        $rows = [];
+
+        foreach ($users as $user) {
+            $rows[] = array_map(
+                static function (string $field) use ($user) {
+                    if (\in_array($field, ['tstamp', 'dateAdded', 'lastLogin'], true)) {
+                        return $user->{$field} ? date('Y-m-d H:i:s', (int) $user->{$field}) : '';
+                    }
+
+                    if ('admin' === $field) {
+                        return $user->{$field} ? '✔' : '';
```

Comment:
Did you make sure that this is displayed correctly on Windows?

Reply:
No... But I remember seeing it somewhere else…
How do I check the checkmark is displayed on Windows? Maybe @fritzmg?

Reply:
It's not really about Windows, but rather about the command line environment used. I expect this to work in the Git bash under Windows for example. I'll test if it works in regular command prompt.

Reply:
Yeah, as expected it does not work in the Windows command prompt. But it does work in the Git bash.

![Annotation 2020-06-24 125304](https://user-images.githubusercontent.com/4970961/85551228-b0827a00-b619-11ea-9409-9cf2705c5e6e.png)

I wouldn't really worry about it. If you use Deployer in the command prompt, you wouldn't see any of its icons either for example.

Reply:
Also note, that the Windows command prompt got some updates in Windows 10. It previously couldn't even display the same color styling 😁 . The whole `composer install` output looked differently. Now it looks the same as under Linux-based bashes.

Reply:
I see. Here is how we have been working around this issue so far: https://github.com/contao/contao/blob/master/manager-bundle/tests/Command/DebugPluginsCommandTest.php#L181

Maybe we should do the same here?

Reply:
Sure, can't hurt.